### PR TITLE
10088 pass staff data to api

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#10088
/bcgov/entity#10047

*Description of changes:*
-finalize staff payment search

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by kialj876 — APPROVED** [review]
Looks good to me as long as Doug is on the same page for how the staff payment is getting sent -- not sure if he's planning to get it from the payload or not

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts [hunk 1]
```diff
@@ -76,4 +76,28 @@ describe('Payment component', () => {
...
+      bcolAccountNumber: '',
+      datNumber: '',
+      folioNumber: '',
+      isPriority: false
+    })
+    await flushPromises()
+    wrapper.findComponent(BaseDialog).vm.$emit(proceed, true)
+    await flushPromises()
+    expect(wrapper.vm.$store.state.stateModel.staffPayment.routingSlipNumber).toBe('999888777')
+  })
```
Comment (kialj876) [inline ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:102]: also would be good to have a test that checks the staff payment info is reset on a new one

File: ppr-ui/src/utils/ppr-api-helper.ts [hunk 2]
```diff
@@ -89,6 +91,29 @@ export async function search (
...
+    extraParams = extraParams + 'certified=True'
+  }
+  // filled out staff payment parameters
+  if (staffPayment) {
+    if (extraParams.length > 0) {
+      extraParams = extraParams + '&'
+    }
+    extraParams = extraParams + 'routingSlipNumber=' + staffPayment.routingSlipNumber
+    extraParams = extraParams + '&bcolAccountNumber=' + staffPayment.bcolAccountNumber
+    extraParams = extraParams + '&datNumber=' + staffPayment.datNumber
```
Comment (doug-lovett) [inline ppr-ui/src/utils/ppr-api-helper.ts]: The parameters are conditional on payment method:
If FAS payment method only include routingSlipNumber.
If BCOL payment only include bcolAccountNumber and datNumber.
If both a routing slip and a bcol account number are submitted the API will return an error.

**Review by doug-lovett — COMMENTED** [review]
See my ppr-api-helper comment. I'm trying to avoid putting it in the payload so the schemas do not need to be modified as they are exposed in the spec and staff payment content may be confusing to external developers.

File: ppr-ui/src/utils/ppr-api-helper.ts (hunk 2 above)
Comment (jordiwes) [reply ppr-ui/src/utils/ppr-api-helper.ts]: Made this change

File: ppr-ui/tests/unit/StaffPaymentDialog.spec.ts (hunk 1 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/StaffPaymentDialog.spec.ts:102]: added a test to search bar (it's where it resets)

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Pull request merged**